backend/internal/controllers/user_test.go

```diff
@@ -39,3 +41,53 @@ func TestCreateDuplicateUser(t *testing.T) {
 	testutils.MakeAuthenticatedRequest("POST", ts.URL+"/users", body, http.StatusBadRequest, &response, "test-token-user1", t)
 	require.Equal(t, response, "user with email already exists")
 }
+
+func TestGetUser(t *testing.T) {
+	var response models.User
+	testutils.MakeAuthenticatedRequest("GET", ts.URL+"/users/user1", "", http.StatusOK, &response, "test-token-user1", t)
+	// assertions with response here
+	require.Equal(t, response.ID, "user1")
+	require.Equal(t, response.Name, "robert")
+	require.Equal(t, response.Email, "test4@test.com")
+	require.Equal(t, response.InSanDiego, true)
+}
+
+func TestGetUserNotFound(t *testing.T) {
+	var response string
+	testutils.MakeAuthenticatedRequest("GET", ts.URL+"/users/user2", "", http.StatusNotFound, &response, "test-token-user2", t)
+	require.Equal(t, response, "user does not exist")
+}
+
+func TestGetUserUnAuth(t *testing.T) {
+	var response string
+	testutils.MakeAuthenticatedRequest("GET", ts.URL+"/users/user1", "", http.StatusForbidden, &response, "test-token-user2", t)
+}
+
+func TestUpdateUser(t *testing.T) {
+	// valid token test
+	body := `{"id": "user1", "name": "roberto", "email": "test4@test.com", "in_san_diego": false}`
```

Comment:
Once you do the above let's check that it actually works by writing two tests that do partial updates. Drop the id and email fields from this request body, and replace it with two tests with bodies: `{"name":"new name"}` and `{"in_san_diego": false}`